opencodeblocks/graphics/view.py

```diff
@@ -221,32 +234,51 @@ def moveViewOnArrow(self, event: QKeyEvent) -> bool:
         for block in code_blocks:
             block_center_x = block.x() + block.width / 2
             block_center_y = block.y() + block.height / 2
-            xdist, ydist = self.getDistanceToCenter(block_center_x, block_center_y)
-            dist_array.append(
-                (
-                    block_center_x,
-                    block_center_y,
-                    xdist,
-                    ydist,
-                    max(abs(xdist), abs(ydist)),
+            if reference is None:
+                xdist, ydist = self.getDistanceToCenter(block_center_x, block_center_y)
+            else:
+                xdist, ydist = (
+                    reference.x() - block_center_x,
+                    reference.y() - block_center_y,
                 )
-            )
+            dist_array.append((block_center_x, block_center_y, -xdist, -ydist))
+
+        def in_region(x, y, key):
+            up_right = x / self.width() - y / self.height() >= 0
+            down_right = x / self.width() + y / self.height() >= 0
+            if key == Qt.Key.Key_Up:
+                return up_right and not down_right
+            if key == Qt.Key.Key_Down:
+                return not up_right and down_right
+            if key == Qt.Key.Key_Left:
+                return not up_right and not down_right
+            if key == Qt.Key.Key_Right:
+                return up_right and down_right
 
-        if key_id == Qt.Key.Key_Up:
-            dist_array = filter(lambda pos: pos[3] > 1, dist_array)
-        if key_id == Qt.Key.Key_Down:
-            dist_array = filter(lambda pos: pos[3] < -1, dist_array)
-        if key_id == Qt.Key.Key_Right:
-            dist_array = filter(lambda pos: pos[2] < -1, dist_array)
-        if key_id == Qt.Key.Key_Left:
-            dist_array = filter(lambda pos: pos[2] > 1, dist_array)
+        key_id = event.key()
+        dist_array = filter(lambda pos: in_region(pos[2], pos[3], key_id), dist_array)
         dist_array = list(dist_array)
-
-        if len(dist_array) <= 0:
+        if len(dist_array) == 0:
             return False
-        dist_array.sort(key=lambda d: d[4])
 
-        block_center_x, block_center_y, _, _, _ = dist_array[0]
+        def oriented_distance(x, y, key):
```

Comment:
Do we need this many cases ?
The blocks were already filtered, so for example,
if `key == Qt.Key.Key_Right`, for example, we know that `x >= 0`.
Maybe an absolute value would be clearer to define a distance.